packages/next/src/server/base-server.ts

```diff
@@ -1160,18 +1160,18 @@ export default abstract class Server<
           if (didRewrite && parsedUrl.pathname) {
             addRequestMeta(req, 'rewroteURL', parsedUrl.pathname)
           }
-          const routeParamKeys = new Set<string>()
-
-          for (const key of Object.keys(parsedUrl.query)) {
-            const value = parsedUrl.query[key]
 
-            normalizeNextQueryParam(key, (normalizedKey) => {
-              if (!parsedUrl) return // typeguard
+          // Normalize all the query params to remove the prefixes.
+          const routeParamKeys = new Set<string>()
+          for (const [key, value] of Object.entries(parsedUrl.query)) {
+            if (typeof value === 'undefined') continue
```

Comment:
Seem this line is new? Do we really need to filter them out or we don't have this case?

Reply:
Technically this is due to the type of the `parsedUrl.query` which is `Record<string, string | string[] | undefined> ` to make up for the fact that we don't have `noUncheckedIndexAccess` enabled for the project yet.